packages/workspace/src/command-line/shared.ts

```diff
@@ -46,18 +46,31 @@ const ig = ignore();
 ig.add(readFileIfExisting(`${appRoot.path}/.gitignore`));
 
 export function printArgsWarning(options: YargsAffectedOptions) {
-  const { files, uncommitted, untracked, base, head } = options;
+  const { files, uncommitted, untracked, base, head, all } = options;
 
   if (
     !files &&
     !uncommitted &&
     !untracked &&
     !base &&
     !head &&
+    !all &&
     options._.length < 2
   ) {
     console.log('Note: Nx defaulted to --base=master --head=HEAD');
   }
+
+  if (all) {
+    console.warn(
```

Comment:
Can we collapse this into 1 warning?

```ts
console.warn(
  `Warning: Running affected:* commands with --all is not meant to be done in any sizable project or in CI as it will result in very slow builds.`
)
console.info('Read about rebuilding and retesting only what is affected here: https://nx.dev/guides/monorepo-affected.')
```

Reply:
I changed to add even more lines to be a bit more like the warning produced when running `serve --prod`.

Shorter lines are much easier to read than long lines.